File: src/com/maddyhome/idea/vim/ex/handler/PreviousTabHandler.java
package com.maddyhome.idea.vim.ex.handler;
import com.intellij.openapi.actionSystem.DataContext;
import com.intellij.openapi.editor.Editor;
import com.maddyhome.idea.vim.VimPlugin;
import com.maddyhome.idea.vim.ex.CommandHandler;
import com.maddyhome.idea.vim.ex.CommandName;
import com.maddyhome.idea.vim.ex.ExCommand;
import org.jetbrains.annotations.NotNull;

public class PreviousTabHandler extends CommandHandler {
    public PreviousTabHandler(){
        super(new CommandName[]{
                new CommandName("tabp", "revious"),
                new CommandName("tabN", "ext")
        }, ARGUMENT_OPTIONAL);
    }

    public boolean execute(@NotNull Editor editor, @NotNull DataContext context, @NotNull ExCommand cmd) {
        VimPlugin.getMotion().moveCaretGotoPreviousTab(editor, context,1 );
        return true;
    }
}

File: src/com/maddyhome/idea/vim/ex/handler/PreviousTabHandler.kt
package com.maddyhome.idea.vim.ex.handler

import com.intellij.openapi.actionSystem.DataContext
import com.intellij.openapi.editor.Editor
import com.maddyhome.idea.vim.VimPlugin
import com.maddyhome.idea.vim.ex.CommandHandler
import com.maddyhome.idea.vim.ex.ExCommand
import com.maddyhome.idea.vim.ex.commands
import com.maddyhome.idea.vim.ex.flags

class PreviousTabHandler : CommandHandler(
        commands("tabp[revious]", "tabN[ext]"),
        flags(ARGUMENT_OPTIONAL, RANGE_FORBIDDEN)
) {
    override fun execute(editor: Editor, context: DataContext, cmd: ExCommand): Boolean {
        VimPlugin.getMotion().moveCaretGotoPreviousTab(editor, context, cmd.argument.toIntOrNull() ?: 0)
        return true
    }
}
